airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -689,28 +688,34 @@ def transform_empty_string_to_none(instance: Any, schema: Any):
 
 class IncrementalRestSalesforceStream(RestSalesforceStream, ABC):
     state_checkpoint_interval = 500
-    STREAM_SLICE_STEP = 30
     _slice = None
 
-    def __init__(self, replication_key: str, **kwargs):
+    def __init__(self, replication_key: str, stream_slice_step: str = "P30D", **kwargs):
         super().__init__(**kwargs)
         self.replication_key = replication_key
+        self._stream_slice_step = stream_slice_step
```

Comment:
Can we ensure that the `stream_slice_step` is big enough so that the step is less granular than the salesforce API? So for example, if Salesforce supports seconds as the smallest increment, `stream_slice_step` should be greater or equal to a second

Reply:
Salesforce stores datetime `fields` in YYYY-MM-DDThh:mm:ssZ [format](https://developer.salesforce.com/docs/atlas.en-us.248.0.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select_dateformats.htm),
so i think min step should be >= 1 second.

Reply:
tests added